src/main/ApplicationImpl.cpp

```diff
@@ -108,7 +108,7 @@ ApplicationImpl::initialize()
     mBucketManager = BucketManager::create(*this);
     mCatchupManager = CatchupManager::create(*this);
     mHistoryManager = HistoryManager::create(*this);
-    mInvariantManager = InvariantManager::create(*this);
+    mInvariantManager = createInvariantManager();
```

Comment:
We should start thinking of injecting these dependencies into application more seriously. Maybe this PR could start that with passing InvariantManager to Application?